File: abc087b.cpp
#include <bits/stdc++.h>
using namespace std;

int A, B, C, X;
int sum, ans;

int main(){

    cin >> A >> B >> C >> X;
    sum = 0;
    ans = 0;

    for (int i=0; i<A+1; i++){
        for (int j=0; j<B+1; j++){
            for (int k=0; k<C+1; k++){
                sum = i*500 + j*100 + k*50;
                if (sum == X) ans++;
                sum = 0;
            }
        }
    }

    cout << ans << endl;
    return 0;
}

File: atcoder/abc087b.py
A, B, C, X = [int(input()) for i in range(4)]

sum = 0
ans = 0

for i in range(A+1):
    for j in range(B+1):
        for k in range(C+1):
            sum = i*500 + j*100 + k*50
            
            if sum == X:
                ans += 1

print(ans)